ts/src/huobi.ts

```diff
@@ -2931,27 +3026,40 @@ export default class huobi extends Exchange {
 
     networkIdToCode (networkId, currencyCode = undefined) {
         // here network-id is provided as a pair of currency & chain (i.e. trc20usdt)
-        const keys = Object.keys (this.options['networkNamesByChainIds']);
-        const keysLength = keys.length;
-        if (keysLength === 0) {
-            throw new ExchangeError (this.id + ' networkIdToCode() - markets need to be loaded at first');
+        if (!('networkNamesByChainIds' in this.options)) {
+            if (!this.currencies) {
+                // currencies need to be loaded, otherwise it's impossible to work
+                throw new ExchangeError (this.id + ' networkIdToCode() - markets need to be loaded at first');
+            }
+            this.options['networkNamesByChainIds'] = {};
+            const currencyCodes = Object.keys (this.currencies);
+            for (let i = 0; i < currencyCodes.length; i++) {
+                const code = currencyCodes[i];
+                const networkValues = this.values (this.currencies[code]['networks']);
+                for (let j = 0; j < networkValues.length; j++) {
+                    const network = networkValues[j];
+                    const currentNetworkId = network['id'];
+                    const currentNetworkCode = network['network'];
+                    this.options['networkNamesByChainIds'][currentNetworkId] = currentNetworkCode;
+                }
+            }
         }
-        const networkTitle = this.safeValue (this.options['networkNamesByChainIds'], networkId, networkId);
-        return super.networkIdToCode (networkTitle);
+        const networkCode = this.safeValue (this.options['networkNamesByChainIds'], networkId, networkId);
+        return super.networkIdToCode (networkCode, currencyCode);
```

Comment:
also this method is not working as expected 
![image](https://github.com/ccxt/ccxt/assets/43336371/1abb554f-5974-464e-9650-f85ffa4bf285)

Reply:
this method only works after currencies are loaded, without it it does not know which network is `usdterc20`

Reply:
@ttodua our cli loads the markets first so currencies are loaded